examples/replay.cc

```diff
@@ -42,24 +45,78 @@ class Replay : public sc2::ReplayObserver {
         }
         std::cout << "Finished" << std::endl;
     }
+
+    //Given metadata about a replay, determine if it meets the criteria for replaying.
+    //Common criteria: high mrr games, certain maps, certain match ups
+    void IgnoreReplay(const ReplayInfo &replay_info, uint32_t &player_id)
+    {
+        //filter based on number of players 
+        if (replay_info.num_players != 2)
+        {
+            return true;
+        }
+
+        //check stats about each individual player in the game
+        for (unsigned int i = 0; i < replay_info.num_players; i++)
+        {
+            //filter based on mrr
+            if (replay_info.players[i].mmr < 3800)
+            {
+                return true;
+            }
+
+            //filter based on match up
+            //This will cause only terran vs terran replays
+            if (replay_info.players[i].race == sc2::Terran)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        //filter by map
+        if (replay_info.map_name == sc2::kMapBelShirVestigeLE)
+        {
+            return true;
+        }
+        
+        //if all the tests passed, observe the replay
+        return false;
+    }
 };
 
 
 int main(int argc, char* argv[]) {
     sc2::Coordinator coordinator;
+
+    //Make sure to pass the appropriate arguemnts when running the program.
     if (!coordinator.LoadSettings(argc, argv)) {
         return 1;
     }
 
+    //load in the replay file or folder or replays
     if (!coordinator.SetReplayPath(kReplayFolder)) {
-        std::cout << "Unable to find replays." << std::endl;
+        std::cout << "Unable to load replay(s)." << std::endl;
         return 1;
     }
 
-    Replay replay_observer;
+    //Keep processing replays until there are none left to process
+    while (coordinator.HasReplays())
```

Comment:
Loop here isn't really necessary. coordinator.Update() will keep returning true until it is finished with all replays.